components/script/dom/htmlinputelement.rs

```diff
@@ -354,8 +360,14 @@ impl HTMLInputElementMethods for HTMLInputElement {
                             |a| DOMString::from(a.summarize().value))
             }
             ValueMode::Filename => {
-                // TODO: return C:\fakepath\<first of selected files> when a file is selected
-                DOMString::from("")
+                let default = DOMString::from("");
+                match *self.filelist.borrow() {
+                    Some(ref fl) => match fl.Item(0) {
```

Comment:
Ahhh it my fault. I thought the comment doesn't mean fakepath as it is .... http://www.w3.org/TR/html5/forms.html#dom-input-value-filename